face-landmarker/index.ts

```diff
@@ -186,40 +107,49 @@ function enableCam(event) {
     enableWebcamButton.innerText = "DISABLE PREDICTIONS";
   }
 
-  // getUsermedia parameters.
-  const constraints = {
-    video: true
-  };
-
   // Activate the webcam stream.
-  navigator.mediaDevices.getUserMedia(constraints).then(function (stream) {
+  navigator.mediaDevices.getUserMedia({video: true}).then(function (stream) {
     video.srcObject = stream;
-    video.addEventListener("loadeddata", predictWebcam);
+    video.addEventListener("loadeddata", detectLoop);
   });
 }
 
 let lastVideoTime = -1;
-let results = undefined;
-const drawingUtils = new DrawingUtils(canvasCtx);
-async function predictWebcam() {
-  const radio = video.videoHeight / video.videoWidth;
-  video.style.width = videoWidth + "px";
-  video.style.height = videoWidth * radio + "px";
-  canvasElement.style.width = videoWidth + "px";
-  canvasElement.style.height = videoWidth * radio + "px";
-  canvasElement.width = video.videoWidth;
-  canvasElement.height = video.videoHeight;
-  // Now let's start detecting the stream.
-  if (runningMode === "IMAGE") {
-    runningMode = "VIDEO";
-    await faceLandmarker.setOptions({ runningMode: runningMode });
+const detectLoop = () => {
+  if (!webcamRunning) {
+    return;
   }
-  let nowInMs = Date.now();
-  if (lastVideoTime !== video.currentTime) {
+  if (video.currentTime !== lastVideoTime) {
     lastVideoTime = video.currentTime;
-    results = faceLandmarker.detectForVideo(video, nowInMs);
+
+    const ratio = video.videoHeight / video.videoWidth;
+    video.style.width = `${videoWidth}px`;
+    video.style.height = `${videoWidth * ratio}px`;
+    
+    const imageData = createImageData(video, videoWidth, videoWidth * ratio);
+    worker.postMessage({
+      mode: 'VIDEO',
```

Comment:
What is the mode used for? If this is the ImageMode then we also need to send the timestap.

Reply:
I updated video detection with timestamp